lisa/notifier.py

```diff
@@ -198,10 +198,14 @@ def notify(message: MessageBase) -> None:
                 if len(_message_queue) > 0:
                     current_message = _message_queue.pop()
             if current_message:
-                notifiers = _messages.get(type(current_message))
-                if notifiers:
+                message_types = type(current_message).__mro__
+                for message_type in message_types:
+                    notifiers = _messages.get(message_type, [])
```

Comment:
Wouldn't this change result in calling notifier unnecessarily for the parent class of a Message?

Reply:
This method is called when a new message arrives. And this loop is to check if there is any registered notifier on each inherited message types. If no notifier registers, there is no further action. It's used for handle message type from a super class, like PerfMessage.